### DISCUSSION TITLE
Rename SqlType(X) as JdbcType(X)

### DISCUSSION THREAD

**Discussion by sebersole** [hibernate/hibernate-orm#3856] [Design Proposals]
The premise of this change is to better communicate the fact that `SqlTypeDescriptor` really describes a "JDBC type".

Rename:
* `org.hibernate.type.descriptor.sql.SqlTypeDescriptor` -> `org.hibernate.type.descriptor.sql.JdbcTypeDescriptor`
* `org.hibernate.type.descriptor.sql.SqlTypeDescriptorIndicators` -> `org.hibernate.type.descriptor.sql.JdbcTypeDescriptorIndicators`
* `org.hibernate.type.descriptor.sql.spi.SqlTypeDescriptorRegistry` -> `org.hibernate.type.descriptor.sql.spi.JdbcTypeDescriptorRegistry`
* `org.hibernate.type.SqlTypeDescriptorIndicatorCapable` -> `org.hibernate.type.JdbcTypeDescriptorIndicatorCapable`
* `@SqlType` -> `@JdbcType`
* `@SqlTypeCode` -> `@JdbcTypeCode`
* `@SqlTypeRegistration` -> `@JdbcTypeRegistration`

Possibly rename package `org.hibernate.type.descriptor.sql` to `org.hibernate.type.descriptor.jdbc`.

**Comment by gavinking** [comment]
Double thumbs up :)

**Comment by beikov** [comment]
This is already done, so I'll close this.
